src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package Duke;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The Parser class is responsible for parsing user input and generating corresponding commands.
+ */
+public class Parser {
+    private static final Scanner scanner = new Scanner(System.in);
```

Comment:
Since this is a constant I believe the name should be in screaming snake case i.e SCANNER. Similar issues elsewhere